ledger/artifactmanager/artifactmanager.go

```diff
@@ -232,17 +233,54 @@ func (m *LedgerArtifactManager) DeployCode(
 	code []byte,
 	machineType core.MachineType,
 ) (*core.RecordID, error) {
-	return m.setRecord(
-		&record.CodeRecord{
-			SideEffectRecord: record.SideEffectRecord{
-				Domain:  domain,
-				Request: request,
+	pulseNumber, err := m.db.GetLatestPulseNumber()
+	if err != nil {
+		return nil, err
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+	var setRecord *core.RecordID
+	var setRecordErr error
+	go func() {
+		id, setRecordErr := record.CalculateIDForBlob(pulseNumber, code)
+		if setRecordErr != nil {
+			wg.Done()
+			return
+		}
+
+		setRecord, setRecordErr = m.setRecord(
+			&record.CodeRecord{
+				SideEffectRecord: record.SideEffectRecord{
+					Domain:  domain,
+					Request: request,
+				},
+				Code:        id,
+				MachineType: machineType,
 			},
-			Code:        code,
-			MachineType: machineType,
-		},
-		request,
-	)
+			request,
+		)
+		wg.Done()
+	}()
+
+	var setBlobErr error
+	go func() {
```

Comment:
goroutine no needed here

Reply:
It's needed here, we must run 2 'parallel' requests